site/content/contribute/more-info/server/style-guide.md

```diff
@@ -179,6 +247,31 @@ func (a *App) SendNotifications(...) {
 }
 ```
 
+#### Avoid double-logging
+
+Double-logging is when you immediately log something after an error, and also return an error at the same time. This creates two log lines with the same error and is confusing to the admin.
+
+For example:
+
+```go
+if err != nil {
+	mlog.Error("Failed to generate SQL query",
+		mlog.String("user_id", userID),
+		mlog.Int("timestamp", int(syncTime)),
+		mlog.Err(err),
+	)
+	return errors.Wrap(err, "failed to generate SQL query")
+}
+```
+
+A better way is:
```

Comment:
The important part is that the caller is responsible for handling the error. @agnivade, you wrote a great summary in one of your reviews that made that clear. Maybe we can transfer some of that to this document?

Reply:
Found it! https://github.com/mattermost/enterprise/pull/1905#discussion_r2067858593